src/api/api_client.rs

```diff
@@ -168,6 +168,18 @@ where
 		})
 	}
 
+	pub fn update_runtime(mut self) -> ApiResult<Self> {
+		let metadata = Self::get_metadata_from_node(&self.client).map(Metadata::try_from)??;
+		debug!("Metadata: {:?}", metadata);
+
+		let runtime_version = Self::get_runtime_version_from_node(&self.client)?;
+		info!("Runtime Version: {:?}", runtime_version);
+
+		self.metadata = metadata;
+		self.runtime_version = runtime_version;
+		Ok(self)
+	}
```

Comment:
I think this should be `pub fn update_runtime(&mut self) -> ApiResult<Metadata>`. The `mut self` is rather used for the builder pattern and not the object itself (btw. I think the `setSigner` method should also be changed).

I don't now if we want to return anything at all, it could also just be `()`

Reply:
Agreed. Tried to be consistent, with what's there already, but passing a mut reference makes more sense.

Reply:
Done :)